src/modules/auth/component/SignUp/SignUp.tsx

```diff
@@ -1,58 +1,62 @@
+/* eslint-disable no-console */
 import {
   Grid,
+  Typography,
 } from "@mui/material";
-import {
-  ChangeEvent, FormEvent, useEffect, useState,
-} from "react";
+import { useEffect } from "react";
 import { v4 as uuidv4 } from "uuid";
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
 import { PrimaryButton } from "../../../../components/UI/PrimaryButton";
 import { TextField } from "../../../../components/UI/TextField";
 import { Heading } from "../../../../components/UI/Heading";
-import { profileData, token } from "../../../../data-mock/data";
+import { token } from "../../../../data-mock/data";
 import { Profile } from "../../../../types/types";
 import { useAppDispatch } from "../../../../store/hooks";
 import { login } from "../../../../store/features/tokenSlice";
-import { UserRepository, userRepository } from "../../../../data-mock/userMock";
+import { User, UserRepository, userRepository } from "../../../../data-mock/userMock";
 
 export function SignUp() {
-  const [formData, setFormData] = useState<Omit<Profile, "id" | "isLoggedIn">>(
-    profileData,
-  );
+  const {
+    control, setError, handleSubmit, formState: { errors }, reset, getValues,
+  } = useForm({
+    defaultValues: {
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+    },
+  });
+
   const users: UserRepository = userRepository;
   const dispatch = useAppDispatch();
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const { name, value } = e.target;
-    setFormData({
-      ...formData,
-      [name]: value,
-    });
-  };
-
   useEffect(() => {
     // eslint-disable-next-line no-console
     console.log("All Submitted Data:", users);
   }, [users]);
 
-  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    const newFormData: Profile = {
-      id: uuidv4(),
-      ...formData,
-    };
-
-    await users.addUser(newFormData);
-    dispatch(login(token));
-    localStorage.setItem("user", JSON.stringify({ email: newFormData.email }));
-    localStorage.setItem("token", token);
-    setFormData(
-      {
-        name: "",
-        email: "",
-        password: "",
-        confirmPassword: "",
-      },
-    );
+  const onSubmit: SubmitHandler<Omit<Profile, "id">> = async (data) => {
+    const userExists: User | null = await users.getUserByEmail(data.email);
+    if (userExists) {
```

Comment:
Use early return instead of else
```
if (userExists) {
  setError("email", { type: "manual", message: "User with such email is already registered" });
  return;
}

const newFormData: Profile = {
        id: uuidv4(),
        ...data,
      };
      await users.addUser(newFormData);
      dispatch(login(token));
      localStorage.setItem("user", JSON.stringify({ email: newFormData.email }));
      localStorage.setItem("token", token);
      reset(
        {
          name: "",
          email: "",
          password: "",
          confirmPassword: "",
     },
  );
```

Reply:
done